src/action/interfaces.rs

```diff
@@ -81,7 +83,16 @@ impl super::Response for Response {
 
     fn into_proto(self) -> Interface {
         Interface {
-            mac_address: Some(mac_to_vec(self.interface.mac_address())),
+            mac_address: match self.interface.mac {
+                Some(mac) => Some(mac_to_vec(mac)),
+                None => {
+                    error!(
+                        "unable to get MAC address for {} interface",
+                        self.interface.name,
+                    );
+                    None
+                },
+            },
```

Comment:
I would assign result of this `match` to a separate variable. Mixing-in imperative logic (printing errors) into structure construction is generally not a good idea.